streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -150,23 +152,46 @@ public void process(final K key, final V value) {
                 return;
             }
 
-            processInOrder(key, value, inputRecordTimestamp, closeTime);
-        }
-
-        public void processInOrder(final K key, final V value, final long inputRecordTimestamp, final long closeTime) {
+            if (reverseIteratorPossible == null) {
+                try {
+                    windowStore.backwardFetch(key, 0L, 0L);
+                    reverseIteratorPossible = true;
+                    log.debug("Sliding Windows aggregate using a reverse iterator");
+                } catch (final UnsupportedOperationException e)  {
+                    reverseIteratorPossible = false;
+                    log.debug("Sliding Windows aggregate using a forward iterator");
+                }
+            }
 
             final Set<Long> windowStartTimes = new HashSet<>();
 
             // aggregate that will go in the current record’s left/right window (if needed)
-            ValueAndTimestamp<Agg> leftWinAgg = null;
-            ValueAndTimestamp<Agg> rightWinAgg = null;
+            final ValueAndTimestamp<Agg> leftWinAgg = null;
+            final ValueAndTimestamp<Agg> rightWinAgg = null;
 
             //if current record's left/right windows already exist
-            boolean leftWinAlreadyCreated = false;
-            boolean rightWinAlreadyCreated = false;
+            final boolean leftWinAlreadyCreated = false;
+            final boolean rightWinAlreadyCreated = false;
 
-            Long previousRecordTimestamp = null;
+            final Long previousRecordTimestamp = null;
```

Comment:
Why create all of these here and then pass the uninitialized values into `processInOrder/processReverse`? If we only need them within the `processX` methods, let's just keep them there